### PR TITLE
[Intel MKL] update mkl addn op for multi-input

### PR DESCRIPTION
**Pull request by guizili0**

### CODE REVIEW

File: tensorflow/core/kernels/mkl_aggregate_ops.cc [hunk 1]
```diff
@@ -38,202 +38,192 @@ class MklAddNOp : public OpKernel {
...
+        if (src_mkl_shape.IsMklTensor()) {
+          md = src_mkl_shape.GetMklLayout();
         } else {
-          output_mkl_shape.SetTfLayout(src2_dims_size,
-                                       src2_mkl_shape.GetSizesAsMklDnnDims(),
-                                       src2_mkl_shape.GetTfDataFormat());
+          if (has_mkl_input) {
+            // MKL input has the data format information.
+            memory::format mkl_data_format = mkl_shape.GetTfDataFormat();
+            auto tf_data_format =
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl_aggregate_ops.cc]: This and the line above can be done after `GetMklShape(ctx, mkl_input_index, &mkl_shape);` in line 167.

File: tensorflow/core/kernels/mkl_aggregate_ops.cc [hunk 2]
```diff
@@ -38,202 +38,192 @@ class MklAddNOp : public OpKernel {
...
+        output_mkl_shape.SetMklLayout(&output_pd);
+        output_mkl_shape.SetElemType(MklDnnType<T>());
+        output_mkl_shape.SetTfLayout(mkl_shape.GetDimension(),
+                                     mkl_shape.GetSizesAsMklDnnDims(),
+                                     mkl_shape.GetTfDataFormat());
         output_tf_shape.AddDim((output_pd.get_size() / sizeof(T)));
       } else {
-        output_mkl_shape.SetMklTensor(false);
-        output_tf_shape = src1_tensor.shape();
+        // all inputs are in Eigen input, get the shape from first one.
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl_aggregate_ops.cc]:
```suggestion
        // All inputs have TF shapes, get the shape from first one.
```

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for your PR! I have minor formatting changes.

File: tensorflow/core/graph/mkl_layout_pass.cc [hunk 3]
```diff
@@ -1651,12 +1637,6 @@ static void FillInputs(const Node* n,
     }
   }
   std::sort(control_edges->begin(), control_edges->end());
-  if (n->op_def().is_commutative()) {
-    // For commutative inputs, we sort the input by the input Node*
-    // to get a canonical ordering (so that add(a,b) and add(b, a) will
-    // hash to the same value if is_commutative is true for 'add').
-    std::sort(in->begin(), in->end());
-  }
```
[the remark this answers is not in the document]
Comment (guizili0) [reply tensorflow/core/graph/mkl_layout_pass.cc]: For multi input of float add, change input order will have very little influence on the precision, so we remove this sort.

**Comment by guizili0** [comment]
@penpornk I have update my code based on your comments, please help to review it again, thanks.

**Review by penpornk — APPROVED** [review]
Thank you for the fixes and the clarifications!

**Pull request merged**